frontend/components/Domain/Recipe/RecipePage/RecipePageParts/RecipePageInstructionsTimer.vue

```diff
@@ -0,0 +1,191 @@
+<template v-if="timers && timers.length">
+  <div @click.stop>
+
+    <v-divider class="mb-2"></v-divider>
+    <div
+      v-for="(timer, i) in timers"
+      :key="i"
+      class="d-flex align-center my-2 justify-center">
+      <v-icon :color="timer.done.value ? 'success' : ''" :class="timer.done.value ? 'shake' : ''">mdi-timer</v-icon>
+      <v-btn
+        icon
+        :disabled="timer.remainingTime.value <= 30"
+        depressed
+        @click="timer.remainingTime.value -= 30"><v-icon>mdi-minus</v-icon></v-btn>
+      {{ timer.hours.value ? timer.hours.value + "h" : "" }} {{ timer.minutes.value ? timer.minutes.value + "m" : "" }}
+      {{ (timer.seconds.value || !timer.minutes.value) ? timer.seconds.value + "s" : "" }}
+      <v-btn
+        icon
+        depressed
+        @click="timer.remainingTime.value += 30"><v-icon>mdi-plus</v-icon></v-btn>
+      <v-btn
+        v-if="!timer.isCounting.value && !timer.hasStarted.value"
+        rounded
+        depressed
+        @click="timer.startAlarm"> {{$t("recipe.timer.start")}} </v-btn>
+      <v-btn
+        v-else-if="(timer.hasStarted.value && timer.isCounting.value)"
+        rounded
+        depressed
+        @click="timer.pauseTimer"> {{$t("recipe.timer.pause")}} </v-btn>
+      </v-btn>
+      <span v-else-if="!timer.done.value">
+        <v-btn
+          rounded
+          depressed
+          @click="timer.startAlarm">
+          {{$t("recipe.timer.continue")}}
+        </v-btn>
+        <v-btn
+          icon
+          @click="timer.resetTimer"><v-icon>mdi-restore</v-icon></v-btn>
+      </span>
+      <span v-else>
+        <v-btn
+          icon
+          @click="timer.resetTimer"><v-icon>mdi-restore</v-icon></v-btn>
+      </span>
+    </div>
+  </div>
+</template>
+<script lang="ts">
+  import { computed, ref, onBeforeUnmount } from "vue";
+import { useContext } from "@nuxtjs/composition-api";
+
+  import { TranslateResult } from "vue-i18n"
+  // @ts-ignore typescript can't find our audio file, but it's there!
+  import timerAlarmAudio from "~/assets/audio/kitchen_alarm.mp3";
+  export default {
+    props: {
+      text: {
+        type: String,
+        default: "",
+      },
+    },
+    setup(props) {
+      const { i18n } = useContext();
+      // Makes regex work with multiple languages via locales
+      const minuetStrings = computed<string[]>(() => {
+        console.log(i18n.availableLocales);
+        const result: TranslateResult[] = [];
+        i18n.availableLocales.forEach((locale) => {
+          result.push(i18n.t("recipe.timer.minute", locale ))
+          result.push(i18n.t("recipe.timer.minutes", locale ))
+        })
+        return result
+      })
```

Comment:
Love this. Might be worth considering hours and seconds, too

Reply:
Yep, if we're inferring time rather than a user configuring it, we need to handle all cases.

Reply:
Would it maybe make sense to parse instructions upfront on the backend and store them in the database? That way we could leverage a few different parsing methods, while also enabling users to manually add timers directly.

For instance, on import, for users with OpenAI integration, we can add an additional processing step which lets an LLM parse instructions for timers.

There's also [this package](https://pypi.org/project/timelength/) which enables time duration parsing (and has support for custom localizations)

Reply:
That would absolutely be a option. Sadly i don't think I have the skills to implement a solution like this.
Possible to start with simpler solution and add better parsing later? This way we can get a feel if more complex parsing Is needed. 
The python package seams like a good middle ground.

Reply:
Yeah if we go the backend route we can keep it simple for now:
1. Add a database entry to store timers on recipe instructions (we can keep it simple and store them as strings in ISO 8061 duration format)
2. When a recipe is created, parse instructions for time strings (any multi-lingual method works, such as that Python package, or any other package)

And then on database migration we can parse all existing recipes and add missing time duration strings

Reply:
How could we handle locales? The python package supports including a json file, maybe we can read a key from the i18n locales?